oasislmf/pytools/common/data.py

```diff
@@ -141,3 +156,183 @@ def load_as_array(dir_path, name, _dtype, must_exist=True):
 @nb.njit(cache=True)
 def almost_equal(a, b):
     return abs(a - b) < float_equal_precision
+
+
+def read_event_rates(run_dir, filename=EVENTRATES_FILE):
+    """Reads event rates from a CSV file
+    Args:
+        run_dir (str | os.PathLike): Path to input files dir
+        filename (str | os.PathLike): event rates csv file name
+    Returns:
+        unique_event_ids (ndarray[oasis_int]): unique event ids
+        event_rates (ndarray[oasis_float]): event rates
+    """
+    try:
+        event_rate_file = Path(run_dir, filename)
+        data = load_as_ndarray(run_dir, filename[:-4], np.dtype([('event_id', oasis_int), ('rate', oasis_float)]),
+                               col_map={"event_id": "EventIds", "rate": "Event_rates"})
+        if data is None or data.size == 0:
+            logger.info(f"Event rate file {event_rate_file} is empty, proceeding without event rates.")
+            return np.array([], dtype=oasis_int), np.array([], dtype=oasis_float)
+        unique_event_ids = data['event_id']
+        event_rates = data['rate']
+
+        # Make sure event_ids are sorted
+        sort_idx = np.argsort(unique_event_ids)
+        unique_event_ids = unique_event_ids[sort_idx]
+        event_rates = event_rates[sort_idx]
+        return unique_event_ids, event_rates
+    except Exception as e:
```

Comment:
load_as_ndarray has a must_exist parameter that handle the case where the files are missing.
So I think we should just allow missing but remove the try except to avoid ignoring other potential issue that should raise exception.